openquake/risklib/connectivity.py

```diff
@@ -53,77 +122,609 @@ def analysis(dstore):
+
+        # If demand nodes are damaged itself (Example, building collapsed where
+        # demand node is considered)
+
+        ccl_table.loc[~ccl_table.index.isin(extant_demand_nodes), 'CN'] = 0
+        pcl_table.loc[~pcl_table.index.isin(extant_demand_nodes), 'NS'] = 0
+        wcl_table.loc[~wcl_table.index.isin(extant_demand_nodes), 'WS'] = 0
+        eff_table.loc[~eff_table.index.isin(extant_eff_nodes), 'Eff'] = 0
+
+        # To check the the values for each node after the earthquake event
+        # Complete connectivity loss
+        ccl_table['CNS'] = [
+            1 if any(nx.has_path(G, j, i) for j in extant_source_nodes) else 0
+            for i in extant_demand_nodes]
+
+        # Partial Connectivity Loss
+        pcl_table.loc[extant_demand_nodes, 'NS'] = [
+            sum(nx.has_path(G, j, i) for j in extant_source_nodes)
+            for i in extant_demand_nodes]
+
+        wcl_table = calc_weighted_connectivity_loss(
+            G, att, extant_source_nodes, extant_demand_nodes, wcl_table,
+            pcl_table, 'WS', 'NS')
+
+        eff_table = calc_efficiency(G, N, att, eff_table, 'Eff')
+
+        # Connectivity Loss for each node
+        pcl_table['PCL_node'] = 1 - (pcl_table['NS']/pcl_table['NS0'])
+        wcl_table['WCL_node'] = 1 - (wcl_table['WS']/wcl_table['WS0'])
+
+        # Computing the mean of the connectivity loss to consider the overall
+        # performance of the area (at global level)
+        CCL_per_event = 1 - ((ccl_table['CNS'].sum())/ccl_table['CNO'].sum())
+        PCL_mean_per_event = pcl_table['PCL_node'].mean()
+        WCL_mean_per_event = wcl_table['WCL_node'].mean()
+        Glo_eff0_per_event = eff_table['Eff0'].mean()
+        Glo_eff_per_event = eff_table['Eff'].mean()
+        # Calculation of Efficiency loss
+        Glo_effloss_per_event = (
+            Glo_eff0_per_event - Glo_eff_per_event)/Glo_eff0_per_event
+
+        # Storing the value of performance indicators for each event
+        event_connectivity_loss_ccl = event_connectivity_loss_ccl.append(
+            {'event_id': event_id, 'CCL': CCL_per_event}, ignore_index=True)
+        event_connectivity_loss_pcl = event_connectivity_loss_pcl.append(
+            {'event_id': event_id, 'PCL': PCL_mean_per_event},
+            ignore_index=True)
+        event_connectivity_loss_wcl = event_connectivity_loss_wcl.append(
+            {'event_id': event_id, 'WCL': WCL_mean_per_event},
+            ignore_index=True)
+        event_connectivity_loss_eff = event_connectivity_loss_eff.append(
+            {'event_id': event_id, 'EFFLoss': Glo_effloss_per_event},
+            ignore_index=True)
+
+        # To store the sum of performance indicator at nodal level to calulate
+        # the average afterwards
+        ccl_table['CCL_node'] = 1 - ccl_table['CNS']
+        ccl_table1 = ccl_table.drop(columns=['CNO', 'CNS'])
+        ccl_table1 = ccl_table1.reset_index()
```

Comment:
The two lines 
```py
ccl_table1 = ccl_table.drop(columns=['CNO', 'CNS'])
ccl_table1 = ccl_table1.reset_index()
```
can be replaced by
```py
ccl_table.drop(columns=['CNO', 'CNS'], inplace=True).reset_index(inplace=True)
```
so we avoid creating more dataframes. Same goes for the PCL, WCL, and EFF dataframes below.